src/Rate/Rate.tsx

```diff
@@ -0,0 +1,229 @@
+    this.handleChangeValue(index, event);
+  };
+
+  handleKeyDown = (index: number, event: React.KeyboardEvent<HTMLElement>) => {
+    const { keyCode } = event;
+    const { max, allowHalf } = this.props;
+    const { characterValue } = this.state;
+    let nextValue = transformCharacterValueToValue(characterValue);
+
+    if (keyCode === KEY_CODE.RIGHT && nextValue < max) {
+      nextValue = allowHalf ? nextValue + 0.5 : nextValue + 1;
+    } else if (keyCode === KEY_CODE.LEFT && nextValue > 0) {
+      nextValue = allowHalf ? nextValue - 0.5 : nextValue - 1;
+    }
+
+    if (keyCode === KEY_CODE.ENTER) {
+      this.handleChangeValue(index, event);
+    }
+
+    this.setState({ characterValue: this.getCharacterValue(nextValue) });
+  };
+
+  handleFirstMouseEnter = _.throttle(index => {
+    if (this.props.allowHalf) {
+      const newCharacterValue = _.map(this.state.characterValue, (_item, itemIndex) => {
+        if (itemIndex === index) {
+          return 0.5;
+        }
+        return itemIndex < index ? 1 : 0;
+      });
+      this.setState({ characterValue: newCharacterValue });
+    } else {
+      this.handleSecondMouseEnter(index);
+    }
+  }, 100);
+
+  handleSecondMouseEnter = _.throttle(index => {
+    const newCharacterValue = _.map(this.state.characterValue, (_item, itemIndex) =>
+      index < itemIndex ? 0 : 1
+    );
+    this.setState({ characterValue: newCharacterValue });
+  }, 100);
+
+  render() {
+    const {
+      character,
+      className,
+      classPrefix,
+      disabled,
+      max,
+      renderCharacter,
+      vertical,
+      ...rest
+    } = this.props;
+
+    const { characterValue } = this.state;
+    const hoverValue = transformCharacterValueToValue(characterValue);
+
+    const characterWrapperClass = classNames(classPrefix, className, {
+      [this.addPrefix('disabled')]: disabled
+    });
+
+    const unhandled = getUnhandledProps(Rate, rest);
+
+    const plaintextRate = (
+      <div {...unhandled} className={className}>{`${this.getValue()}(${max})`}</div>
+    );
+
+    const rate = (
+      <ul {...unhandled} className={characterWrapperClass} onMouseLeave={this.handleMouseLeave}>
+        {_.map(characterValue, (item, index) => (
+          <li
+            key={index}
+            className={classNames(this.addPrefix('character'), {
+              [this.addPrefix('character-full')]: item === 1,
+              [this.addPrefix('character-half')]: item === 0.5,
+              [this.addPrefix('character-empty')]: item === 0
+            })}
+            tabIndex={0}
+            onClick={!disabled && _.partial(this.handleClick, index)}
+            onKeyDown={!disabled && _.partial(this.handleKeyDown, index)}
+          >
+            <div
+              className={classNames(this.addPrefix('character-first'), {
+                [this.addPrefix('character-vertical')]: vertical
+              })}
+              onMouseMove={!disabled && _.partial(this.handleFirstMouseEnter, index)}
```

Comment:
Warning: Expected `onMouseMove` listener to be a function, instead got `false`.
If you used to conditionally omit it with onMouseMove={condition && value}, pass onMouseMove={condition ? value : undefined} instead.